django/contrib/gis/geos/geometry.py

```diff
@@ -174,12 +174,14 @@ def from_gml(cls, gml_string):
     def __eq__(self, other):
         """
         Equivalence testing, a Geometry may be compared with another Geometry
-        or a WKT representation.
+        or an EWKT representation.
         """
         if isinstance(other, six.string_types):
-            return self.wkt == other
+            if other.startswith('SRID=0;'):
+                return self.wkt == other[7:]  # Test only WKT part of other
```

Comment:
This works:
```python
>>> Point(0, 0, srid=4326) == 'SRID=0;POINT (0 0)'
True
```
This doesn't work, because of the comma:
```python
>>> Point(0, 0, srid=4326) == 'SRID=0;POINT (0, 0)'
True
```
Is that what you mean?